src/Toolkit/Toolkit.Maui/SymbolDisplay/SymbolDisplay.cs

```diff
@@ -176,4 +178,28 @@ private async void Refresh()
     /// </summary>
     [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
     public event System.EventHandler? SourceUpdated;
+
+    private void SymbolDisplay_PropertyChanged(object? sender, System.ComponentModel.PropertyChangedEventArgs e)
+    {
+        if (e.PropertyName == nameof(Window) && Window != null)
+        {
+            Refresh();
+            Window.DisplayDensityChanged += Window_DisplayDensityChanged;
+        }
+    }
+
+    private void SymbolDisplay_Unloaded(object? sender, EventArgs e)
+    {
+        if (Window != null)
+        {
+            Window.DisplayDensityChanged -= Window_DisplayDensityChanged;
+        }
+        this.Unloaded -= SymbolDisplay_Unloaded;
+        this.PropertyChanged -= SymbolDisplay_PropertyChanged;
```

Comment:
You should only unhook here if you hook up in loaded. Since you're doing it in the constructor though, this would mean load->unload->load would not work. I don't think you need to unhook though - the event handler goes out of scope with the control.